packages/neon/neon/lib/src/settings/models/nextcloud_app_options.dart

```diff
@@ -1,23 +1,66 @@
+import 'package:meta/meta.dart';
+import 'package:neon/src/settings/models/exportable.dart';
 import 'package:neon/src/settings/models/option.dart';
 import 'package:neon/src/settings/models/options_category.dart';
 import 'package:neon/src/settings/models/storage.dart';
 
-abstract class NextcloudAppOptions {
-  NextcloudAppOptions(this.storage);
+/// Collection of [Option]s.
+abstract class OptionsCollection implements Exportable {
+  /// Creates a new OptionsCollection.
+  OptionsCollection(this.storage);
 
+  /// Storage backend to use.
+  @protected
   final AppStorage storage;
-  late final List<OptionsCategory> categories;
-  late final List<Option> options;
 
+  /// Collection of options.
+  @protected
+  Iterable<Option> get options;
+
+  /// Resets all [options].
+  ///
+  /// Implementers extending this must call super.
+  @mustCallSuper
   void reset() {
     for (final option in options) {
       option.reset();
     }
   }
 
+  /// Disposes all [options].
+  ///
+  /// Implementers extending this must call super.
+  @mustCallSuper
   void dispose() {
     for (final option in options) {
       option.dispose();
     }
   }
+
+  @override
+  MapEntry<String, Object?> export() {
+    final data = Map.fromEntries(options.serialize());
+
+    return MapEntry(storage.id, data);
+  }
+
+  @override
+  void import(final Map<String, Object?> data) {
+    final value = data[storage.id];
+    if (value != null) {
+      options.deserialize(value as Map<String, Object?>);
+    }
+  }
+}
+
+/// OpptionsCollection for a neon app.
```

Comment:
```suggestion
/// OpptionsCollection for an [AppImplementation].
```

Is this better?

Reply:
The thing is that `AppImplementation` is not visible in that scope and just importing it for the reference in the comment doesn't feel right

Reply:
Alright